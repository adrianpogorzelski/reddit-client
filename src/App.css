body {
    background-color: pink;
}

h1, footer, img {
    text-align: center;
}

.flex {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-around;
}