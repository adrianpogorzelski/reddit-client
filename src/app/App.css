body {
    background-image: url("bg.jpg");
    background-size: contain;
}

h1, footer, img {
    text-align: center;
}

#search-bar {
    width: 50em;
    height: 2em;
    margin-top: auto;
    margin-bottom: auto;;
}

button {
    padding: 0.5em;
    margin: 1em;
    border-radius: 5px;
    border: 1px solid black;
}

.flex {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-around;
}

.centered {
    margin-left: auto;
    margin-right: auto;
    justify-content: center;
}