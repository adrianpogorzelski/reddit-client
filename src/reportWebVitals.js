import fetch from "node-fetch";

const reportWebVitals = onPerfEntry => {
  if (onPerfEntry && onPerfEntry instanceof Function) {
    import('web-vitals').then(({ getCLS, getFID, getFCP, getLCP, getTTFB }) => {
      getCLS(onPerfEntry);
      getFID(onPerfEntry);
      getFCP(onPerfEntry);
      getLCP(onPerfEntry);
      getTTFB(onPerfEntry);
    });
  }
};

export default reportWebVitals;
const fetchRedditData = async (subreddit = 'popular') => {
  const url = `https://www.reddit.com/r/${subreddit}.json`;
  const response = await fetch(url)
  return response.json()
}