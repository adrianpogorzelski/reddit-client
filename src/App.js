import './App.css';

function App() {
  return (
    <reddit-client>
      <header>
        <h1>Reddit Client: A Codecamy React/Redux project</h1>
      </header>
      <nav>
        <ul>
          <li>Menu item</li>
          <li>Menu item</li>
          <li>Menu item</li>
          <li>Menu item</li>
        </ul>
      </nav>
      <main>
        <p>Content</p>
      </main>
    </reddit-client>
  );
}

export default App;
