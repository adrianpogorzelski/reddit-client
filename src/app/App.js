import React from 'react';
import './App.css';
import Filters from '../components/filters/filters'

const App = (props) => {
    return (
    <client-for-reddit>
      <header>
        <h1>Reddit Client: A Codecamy React/Redux project</h1>
      </header>
      <main>
        <section className="flex centered">
            <input id="search-bar" type="text" placeholder="Search for a subreddit..."/>
            <button>Search</button>
        </section>
          <Filters />
        <section id="storeRender">
            <h2>Select a category</h2>
            <img src="https://upload.wikimedia.org/wikipedia/commons/2/2a/Loading_Key.gif" alt="Loading" />
        </section>
      </main>
      <footer>
        <p>All data comes from Reddit.</p>
      </footer>
    </client-for-reddit>
  );
}

export default App;
