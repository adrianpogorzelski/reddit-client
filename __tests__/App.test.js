import 'jsdom-global/register'
import React from "react";
import "../src/setupTests"
import {mount, render, shallow} from "enzyme";
import App from "../src/app/App";

let wrapper = shallow(<App/>);

describe("Rendering", () => {
    describe("Main App component", () => {
        test("Renders", () => {
            expect(wrapper).toMatchSnapshot()
        })
    })
    describe("Components", () => {
        describe('Searchbar', () => {
            test("Renders", () => {
                expect(wrapper.find('Searchbar')).toHaveLength(1)
            })
        });
        describe('Filters', () => {
            test("Renders", () => {
                expect(wrapper.find('Filters')).toHaveLength(1)
            })
        });
        // Tiles
            // Are rendered
            // 25 of them
    })
})