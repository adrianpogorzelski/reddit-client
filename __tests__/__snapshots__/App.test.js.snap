// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Main App component Renders 1`] = `
<client-for-reddit
  id="client"
>
  <header>
    <h1>
      Reddit Client: A Codecamy React/Redux project
    </h1>
  </header>
  <main>
    <nav>
      <Searchbar />
      <Filters />
    </nav>
    <section
      id="storeRender"
    >
      <h2>
        Select a category
      </h2>
      <img
        alt="Loading"
        src="https://upload.wikimedia.org/wikipedia/commons/2/2a/Loading_Key.gif"
      />
    </section>
  </main>
  <footer>
    <p>
      All data comes from Reddit.
    </p>
  </footer>
</client-for-reddit>
`;
